emcache/_cython/key_validation.pyx

```diff
@@ -11,8 +13,11 @@ def is_key_valid(bytes key) -> bool:
     if len(key) > MAX_KEY_LENGTH:
         return False
 
-    for c in key:
-        if c < 33 or c > 126:
```

Comment:
Hey, maybe Im missing something but whats the problem on relaxing this with:
```python
if c < 33 or c == 127:
  return false
```

This should make the trick unless Im missing something.

`emcache` is not opinionated on what serialisation are you using for your key and is only demanding just a raw sting of bytes, and enforces the memached requirements for this array of bytes.

I would expect that with that change if you then add a new test case with

```
>>> "ñ".encode()
b'\xc3\xb1'
```

This should work since any of the characters used by the UTF encoder are mapped to the invalid characters.

Reply:
The problem I ran into here is that `for c in key` looped over each individual byte in the bytestring, first evaluating `\xc3` against the condition and then evaluating `\xb1` in the case of ñ. The reason this isn't working for me is that the ordinal number associated with that first byte is -61 which is causing `is_key_valid("ñ".encode("utf8")` to return `False` (because -61 < 33) when it should return `True`. I worked around this by getting the loop to evaluate each character as a whole unit instead of examining individual bytes.

I do see how this enforces utf-8, though, and I understand why that may be undesirable. The only way I can see around that is by adding an encoding argument (that perhaps defaults to utf-8 due to its common usage?)

I'll admit, I'm not super used to Cython, so if I'm still not understanding something, let me know. But I do have an added test case for `"ñ".encode("utf8")` in here already and it passes with this approach but fails if I do the above check.

Reply:
I believe `cdef int c` on line 9 + the fact that many compiles (like gcc) have char as a signed variable by default and that is the culprit of getting -61 < 33 issue.

I think this issue could be resolved in two steps:

1. change `cdef int c` into `cdef unsigned char c`
2. change `if c < 33 or c > 126:` into `if c < 33 or c == 127:`

After that `cyemcache.is_key_valid("ñ".encode("utf8"))` returns `True`.

Reply:
Good catch! if you want to make a PR for changing this!